src/lib/dev-environment/dev-environment-core.js

```diff
@@ -626,23 +641,18 @@ function isVersionListExpired( cacheFile, ttl ) {
 /**
  * Uses a cache file to keep the version list in tow until it is ultimately outdated
  */
-export async function getVersionList() {
+export async function getVersionList(): Promise<WordPressTag[]> {
 	let res;
 	const mainEnvironmentPath = xdgBasedir.data || os.tmpdir();
 	const cacheFile = path.join( mainEnvironmentPath, 'vip', DEV_ENVIRONMENT_WORDPRESS_CACHE_KEY );
 
 	// Handle from cache
 	try {
 		// If the cache doesn't exist, create it
-		if ( ! fs.existsSync( cacheFile ) ) {
-			res = await fetchVersionList();
-			fs.writeFileSync( cacheFile, res );
-		}
-
 		// If the cache is expired, refresh it
-		if ( isVersionListExpired( cacheFile, DEV_ENVIRONMENT_WORDPRESS_VERSION_TTL ) ) {
+		if ( ! fs.existsSync( cacheFile ) || isVersionListExpired( cacheFile, DEV_ENVIRONMENT_WORDPRESS_VERSION_TTL ) ) {
 			res = await fetchVersionList();
-			fs.writeFileSync( cacheFile, res );
+			await fs.promises.writeFile( cacheFile, res );
```

Comment:
Same question as above sync vs promises

Reply:
My personal preference is not to mix async and sync calls. The function already uses `await`, so it makes sense to use the async API.